### PR TITLE
Added a comment to content PR's with a link to the unit test coverage report

### PR DESCRIPTION
**Pull request by MosheEichler**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/43570

## Description
Added a comment with a link to the coveragte unit test report to content PR's via the gitlab pipline (in run unittest and lint job)

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Whois/Integrations/Whois/Whois.py [hunk 1]
```diff
@@ -8512,7 +8512,7 @@ def main():
     if DBotScoreReliability.is_valid_type(reliability):
         reliability = DBotScoreReliability.get_dbot_score_reliability_from_str(reliability)
     else:
-        raise Exception("Please provide a valid value for the Source Reliability parameter.")
+        raise Exception("Please provide a valid value for the Source Reliability parameter1111.")
```
Comment (MosheEichler) [inline Packs/Whois/Integrations/Whois/Whois.py:8515]: This is just for testing to run demisto-sdk lint on some package...

**Review by reutshal — CHANGES_REQUESTED** [review]
The "add_pr_comment" looks very similar to the "_add_pr_comment" in demisto-sdk.
Why not just import it and use it here?

**Comment by MosheEichler** [comment]
> The "add_pr_comment" looks very similar to the "_add_pr_comment" in demisto-sdk. Why not just import it and use it here?

The function is a part of the `test-content` command (and we can't call it directly) And calling the command it's also impossible because we need some value's that is available just from the next jobs for example which test is skipped

File: Tests/scripts/add_pr_comment.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                for existing_comment in issue_comments:
+                    # Check if a comment about report coverage already exists. If there is delete it first and then post
+                    # a new comment:
+                    if COVERAGE_REPORT_COMMENT in existing_comment.get('body'):
+                        comment_url = existing_comment.get('url')
+                        requests.delete(comment_url, headers=headers)
+                requests.post(issue_url, json={'body': f'{COVERAGE_REPORT_COMMENT}:\n {COVERAGE_LINK}'},
+                              headers=headers)
+        else:
+            print('Add pull request comment failed: There is more then one open pull request for branch {branch_name}.')
```
[the remark this answers is not in the document]
Comment (MosheEichler) [reply Tests/scripts/add_pr_comment.py]: No. to be hounest I'm not sure how to check it, how to create two PR's for the same branch.

**Review by reutshal — CHANGES_REQUESTED** [review]

File: Packs/Whois/Integrations/Whois/Whois.py [hunk 3]
```diff
@@ -13,7 +13,7 @@
 # flake8: noqa
 
 """
-    This integration is built using the joepie91 "Whois" module. For more information regarding this package please see
+    aThis integration is built using the joepie91 "Whois" module. For more information regarding this package please see
```
Comment (reutshal) [inline Packs/Whois/Integrations/Whois/Whois.py:16]: Please revert

File: Tests/scripts/add_pr_comment.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+
+COVERAGE_REPORT_COMMENT = 'Link to the unit tests coverage report: \n'
+JOB_ID = os.environ.get("CI_JOB_ID")
+COVERAGE_LINK = f'https://xsoar.docs.pan.run/-/content/-/jobs/{JOB_ID}/artifacts/artifacts/coverage_report/html/' \
+                f'index.html'
+
+
+if __name__ == "__main__":
+    logging_manager = ParallelLoggingManager('Run_Tests.log')
```
Comment (reutshal) [inline Tests/scripts/add_pr_comment.py:13]: This should be a new logger, Run_Tests is not relevant to this flow

File: dev-requirements-py3.txt [hunk 5]
```diff
@@ -1,7 +1,7 @@
 flake8==3.9.2
 bandit==1.7.0
 demisto-py>=3.0.2
-demisto-sdk==1.5.8
+git+https://github.com/demisto/demisto-sdk.git@479d81cb135db9df14fde2134eb7cec78954c644
```
Comment (reutshal) [inline dev-requirements-py3.txt:4]: Don't forget to revert after the testing

**Review by reutshal — CHANGES_REQUESTED** [review]

File: .gitlab/ci/global.yml [hunk 6]
```diff
@@ -205,6 +205,8 @@
           else
             demisto-sdk coverage-analyze -i $ARTIFACTS_FOLDER/coverage_report/.coverage --report-dir $ARTIFACTS_FOLDER/coverage_report --report-type html,xml --previous-coverage-report-url https://storage.googleapis.com/marketplace-dist-dev/code-coverage-reports/coverage-min.json
           fi
+          echo "Adding coverage comment to the pr"
```
Comment (reutshal) [inline .gitlab/ci/global.yml:208]:
```suggestion
          echo "Adding unit tests coverage comment to the pr"
```

File: Tests/scripts/add_pr_comment.py [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+COVERAGE_REPORT_COMMENT = 'Link to the unit tests coverage report: \n'
+JOB_ID = os.environ.get("CI_JOB_ID")
+COVERAGE_LINK = f'https://xsoar.docs.pan.run/-/content/-/jobs/{JOB_ID}/artifacts/artifacts/coverage_report/html/' \
+                f'index.html'
+
+
+if __name__ == "__main__":
+    logging_manager = ParallelLoggingManager('UT_coverage_report.log')
+    _add_pr_comment(COVERAGE_REPORT_COMMENT + COVERAGE_LINK, logging_manager)
```
Comment (reutshal) [inline Tests/scripts/add_pr_comment.py:14]:
```suggestion
JOB_ID = os.environ.get("CI_JOB_ID")
COVERAGE_LINK = f'https://xsoar.docs.pan.run/-/content/-/jobs/{JOB_ID}/artifacts/artifacts/coverage_report/html/' \
                f'index.html'
COVERAGE_REPORT_COMMENT = f'Link to the unit tests coverage report: \n {COVERAGE_LINK}'


if __name__ == "__main__":
    logging_manager = ParallelLoggingManager('UT_coverage_report.log')
    _add_pr_comment(COVERAGE_REPORT_COMMENT, logging_manager)
```

**Review by reutshal — CHANGES_REQUESTED** [review]
Lets change this and I think we are ready to merge

**Comment by MosheEichler** [comment]
Waiting for demisto-sdk version 1.5.9

File: .gitlab/ci/global.yml [hunk 8]
```diff
@@ -205,6 +205,8 @@
           else
             demisto-sdk coverage-analyze -i $ARTIFACTS_FOLDER/coverage_report/.coverage --report-dir $ARTIFACTS_FOLDER/coverage_report --report-type html,xml --previous-coverage-report-url https://storage.googleapis.com/marketplace-dist-dev/code-coverage-reports/coverage-min.json
           fi
+          echo "Adding unit tests coverage comment to the pr"
+          python3 ./Tests/scripts/add_pr_comment.py
```
Comment (esharf) [inline .gitlab/ci/global.yml:209]:
```suggestion
            demisto-sdk coverage-analyze -i $ARTIFACTS_FOLDER/coverage_report/.coverage --report-dir $ARTIFACTS_FOLDER/coverage_report --report-type html,xml --previous-coverage-report-url https://storage.googleapis.com/marketplace-dist-dev/code-coverage-reports/coverage-min.json
              echo "Adding unit tests coverage comment to the pr"
              python3 ./Tests/scripts/add_pr_comment.py

          fi
```

**Review by esharf — APPROVED** [review]
LGTM

**Review by reutshal — APPROVED** [review]

**Pull request merged**